<launch>

  <arg name="debug" default="0"/>
  <arg name="mode" default="default"/>
  <arg name="namespace" default="sam"/>
  
    <group ns="$(arg namespace)">

      <param name="robot_description" 
          command="$(find xacro)/xacro.py '$(find sam_description)/robots/floatsam_usv_$(arg mode).urdf.xacro' debug:=$(arg debug) namespace:=floatsam_usv" />

      <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" respawn="true" output="screen">
        <param name="use_gui" value="false"/>
        <rosparam param="source_list">["/sam/command_states"]</rosparam>
      </node>

      <!-- Publish robot model for ROS -->
      <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="true" output="screen">
        <param name="robot_description" value="/$(arg namespace)/robot_description" />
      </node>
      
      <group ns="dr">

        <node name="gps_pose_pub" pkg="sam_dead_reckoning" type="gps_node.py" respawn="true" output="screen">
          <param name="gps_topic" value="/$(arg namespace)/core/gps" />
          <param name="map_frame" value="map" />
          <param name="utm_frame" value="utm" />
          <param name="gps_frame" value="floatsam_usv/gps_link" />
          <param name="gps_odom_topic" value="/$(arg namespace)/dr/gps_odom_sam" />
        </node>
        
        <node pkg="sam_dead_reckoning" name="dr_node" type="dr_node.py" output="screen">
          <param name="dvl_topic" value="/$(arg namespace)/core/dvl" />
          <param name="dvl_dr_topic" value="/$(arg namespace)/dr/dvl_dr" />
          <param name="stim_imu" value="/$(arg namespace)/core/imu" />
          <param name="sbg_euler" value="/$(arg namespace)/sbg/ekf_euler" />
          <param name="depth_topic" value="/$(arg namespace)/dr/depth"/>                               <!---->
          <param name="pressure_frame" value="floatsam_usv/pressure_link" />
          <param name="odom_frame" value="$(arg namespace)/odom" />
          <param name="base_frame" value="floatsam_usv/base_link" />
          <param name="map_frame" value="map" />
          <param name="dvl_frame" value="$(arg namespace)/dvl_link" />
          <param name="dvl_period" value="0.2" />
          <param name="dr_period" value="0.02" />
        </node>
      
        <include file="$(find auv_particle_filter)/launch/auv_pf.launch">
            <arg name="map_frame" value="map"/>
            <arg name="odom_frame" value="$(arg namespace)/odom"/>
            <arg name="base_frame" value="floatsam_usv/base_link" />
        </include>

      </group>
    </group>

</launch>
